units/recalculate_change_points_test.go

```diff
@@ -0,0 +1,213 @@
+package units
+
+import (
+	"context"
+	"os"
+	"runtime"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/evergreen-ci/cedar/perf"
+
+	"github.com/evergreen-ci/cedar"
+	"github.com/evergreen-ci/cedar/model"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"gopkg.in/mgo.v2/bson"
+)
+
+type TestResultsAndRollups struct {
+	info    *model.PerformanceResultInfo
+	rollups []model.PerfRollupValue
+}
+
+func makePerfResults(length int, breakpoint int, unique string) []TestResultsAndRollups {
+	var rollups []TestResultsAndRollups
+	i := 0
+	for i < length {
+		newRollup := TestResultsAndRollups{
+			info: &model.PerformanceResultInfo{
+				Project:  "project" + unique,
+				Variant:  "variant",
+				Version:  "version" + strconv.Itoa(i+1),
+				Order:    i + 1,
+				TestName: "test",
+				TaskName: "task",
+				Mainline: true,
+			},
+			rollups: []model.PerfRollupValue{
+				{
+					Name:       "measurement",
+					MetricType: model.MetricTypeSum,
+					Version:    0,
+				},
+				{
+					Name:       "measurement_another",
+					MetricType: model.MetricTypeSum,
+					Version:    0,
+				},
+			},
+		}
+		if i < breakpoint {
+			newRollup.rollups[0].Value = float64(100)
+			newRollup.rollups[1].Value = float64(100)
+		} else {
+			newRollup.rollups[0].Value = float64(1000)
+			newRollup.rollups[1].Value = float64(1000)
+		}
+		rollups = append(rollups, newRollup)
+		i++
+	}
+	return rollups
+}
+
+func init() {
+	dbName := "cedar_signal_processing"
+	env, err := cedar.NewEnvironment(context.Background(), dbName, &cedar.Configuration{
+		MongoDBURI:    "mongodb://localhost:27017",
+		DatabaseName:  dbName,
+		SocketTimeout: time.Minute,
+		NumWorkers:    2,
+	})
+	if err != nil {
+		panic(err)
+	}
+	cedar.SetEnvironment(env)
+}
+
+func tearDown(env cedar.Environment) error {
+	conf, session, err := cedar.GetSessionWithConfig(env)
+	if err != nil {
+		return errors.WithStack(err)
+	}
+	defer session.Close()
+	return errors.WithStack(session.DB(conf.DatabaseName).DropDatabase())
+}
```

Comment:
can you use the Client interface rather than the session one? 

additionally we should perhaps call close on the env if we're hoping to clear the entire state of the connection (to be called at the end of the test rather than inbetween tests.)

Reply:
Could you direct me to a test that does this the way you are looking for? I just copied this from `TestFTDCRollupsJob`.

Reply:
I guess it's alright for now.